site/app/templates/course/CourseMaterials.twig

```diff
@@ -297,8 +298,80 @@
     function setNewDateTime(me, path) {
         // pass filename to server to record the new date and time of the file to be released
         var newDateTime = $(me).val();
-
         changeNewDateTime(path, newDateTime);
+        var today = new Date();
+        //to properly compare dates/times they need to match in format
+        //not getting it from the picker leaves out the zero before single digits
+        if(today.getMonth()+1<10){
+            var month = '0'+(today.getMonth()+1);
+        }
+        else{
+            var month = (today.getMonth()+1);
+        }
+        if(today.getDate()<10){
+            var day = '0'+today.getDate();
+        }
+        else{
+            var day = today.getDate();
+        }
+        if(today.getHours()<10){
+            var hours = "0"+today.getHours();
+        }
+        else{
+            var hours = today.getHours();
+        }
+        if(today.getMinutes()<10){
+            var mins = "0"+today.getMinutes();
+        }
+        else{
+            var mins = today.getMinutes();
+        }
+        if(today.getSeconds()<10){
+            var secs = "0"+today.getSeconds();
+        }
+        else {
+            var secs = today.getSeconds();
+        }
+
+        var date = today.getFullYear()+'-'+month+'-'+day;
+
+        var time = hours+":"+mins+":"+secs;
+        var currentDT = date+' '+time;
+        var neverDT = (today.getFullYear()+10)+'-'+month+'-'+day+' '+time;
+
+        //get the value in each file so the color can be assigned
+        //based on the time chosen
+        var fileDT = newDateTime;
+
+        //also custom colors for this page for readability
+        if(fileDT<=currentDT){
+            me.style.backgroundColor = "#87db88";
+
+            return "#87db88";
+        }
+        else if(fileDT>=neverDT){
+            me.style.backgroundColor = "#f76c6c";
+            return "#f76c6c";
+        }
+        else{
+            me.style.backgroundColor = "#ffeb54";
+            return "#ffeb54";
+        }
+    }
+
+    function setChildNewDateTime(path, changeDate,handleData) {
```

Comment:
I'm confused what this function is supposed to do, why can't you directly call `changeFolderNewDateTime` 

```
changeFolderNewDateTime(path, handleData);
```
which already accepts a function callback to pass its response.